ln-gateway/src/ln.rs

```diff
@@ -66,3 +70,66 @@ impl LnRpc for Mutex<cln_rpc::ClnRpc> {
         }
     }
 }
+
+/// Control the behavoir of the LnRpc by exposing the same interface but attach configurations and
+/// methods to it.
+pub struct LnRpcAdapter {
+    // add more configuration
+    ln_client: Box<dyn LnRpc>,
+    fail_times: u8,
+    fail_invoice: Arc<Mutex<HashMap<String, AtomicU8>>>, //does this need to be in arc
```

Comment:
Since `Mutex` is to wrap the whole map, there's no need to use atomics. It actually makes everything slower and harder.

Reply:
Right, first I wanted to just use Atomics and introduced the Mutex later and did not notice that